### DISCUSSION TITLE
Query that filters on property that is a multi-link?

### DISCUSSION THREAD

**Discussion by vedantroy** [geldata/gel#3788] [Q&A] [answered]
Is there a way to do the following in a single query "return a `Foo` object where its multi link to `Bar` objects contains a `Bar` object with the id ABCD"?

**Comment by elprans** [comment, marked as answer]
Sure, something like this?

```
select Foo filter .bar.someId = "ABCD"
```

**Reply by vedantroy** [reply]
I'm trying to figure how to represent this inside of the JS client. The `filter` key only seems to accept booleans? or at least I can't figure out how to represent anything else

**Reply by elprans** [reply]
```
e.select(Foo, foo => ({
  filter: e.op(foo.bar.someId, "=", "ABCD")
})
```

**Reply by candidia** [reply]
I'm trying to do something similar but to filter by where the multi link has all the values in an array I pass in:

I tried:

```js
filter: e.op(
  e.len(
    e.op(
      e.array_unpack(itemNames),
      "union",
      user.items.name
    )
  ),
  "=",
  e.len(user.items.name)
)
```

so filter by the length of the union between the items I'm filtering by and the user's items if that equals the length of the user's items (meaning the user has all the items I'm filtering by). if the size increased, the user would not have one or more of them. This does not work though. Any idea?

**Reply by candidia** [reply]
Never mind, I found out how to:

```js
e.op(
  e.count(
    e.op(
      "distinct",
      e.set(e.array_unpack(itemNames), user.items.name)
    )
  ),
  "=",
  e.count(user.items.name)
)
```

Oddly this use case isn't really documented anywhere I could find so hopefully it helps someone!
